plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -408,6 +413,29 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 deletePredicate);
     }
 
+    private TupleDomain<IcebergColumnHandle> getEffectivePredicate(
+            Schema tableSchema,
+            Map<Integer, Optional<String>> partitionKeys,
+            TupleDomain<IcebergColumnHandle> dynamicFilterPredicate,
+            TupleDomain<IcebergColumnHandle> unenforcedPredicate)
+    {
+        if (dynamicFilterPredicate.isAll() || partitionKeys.isEmpty()) {
+            return unenforcedPredicate.intersect(dynamicFilterPredicate);
+        }
+        Set<IcebergColumnHandle> partitionColumns = partitionKeys.keySet().stream()
+                .map(fieldId -> getColumnHandle(tableSchema.findField(fieldId), typeManager))
+                .collect(toImmutableSet());
+        Supplier<Map<ColumnHandle, NullableValue>> partitionValues = memoize(() -> getPartitionValues(partitionColumns, partitionKeys));
+        if (!partitionMatchesPredicate(partitionColumns, partitionValues, dynamicFilterPredicate)) {
```

Comment:
The unenforced predicate AFAIK does not overlap with partition values.

Reply:
> The unenforced predicate AFAIK does not overlap with partition values.

i think it might, in the case where there are multiple partitioning schemes present in the table